tests/fileVersionTest.js

```diff
@@ -1,35 +1,80 @@
+  const config = require('./config/config.json')
 
   // LIBRARY INSTANCE
-  var oc
+  let oc
 
   // PACT setup
   const Pact = require('@pact-foundation/pact-web')
   const provider = new Pact.PactWeb()
-  const { setGeneralInteractions } = require('./pactHelper.js')
-
-  beforeAll(function (done) {
-    Promise.all(setGeneralInteractions(provider)).then(done, done.fail)
-  })
-
-  afterAll(function (done) {
-    provider.removeInteractions().then(done, done.fail)
-  })
+  const {
+    setGeneralInteractions,
+    validAuthHeaders,
+    origin,
+    accessControlAllowHeaders,
+    accessControlAllowMethods,
+    applicationXmlResponseHeaders
+  } = require('./pactHelper.js')
 
   // TESTING CONFIGS
-  var testFolder = '/testFolder' + timeRightNow
-
-  var versionedFile = testFolder + '/versioned.txt'
-  var versionedFileInfo
+  const versionedFile = config.testFile
+  const fileInfo = {
+    id: 12345678,
+    versions: [
+      {
+        versionId: 98765432,
+        content: '**'
+      },
+      {
+        versionId: 87654321,
+        content: '*'
+      }
+    ]
+  }
+  const propfindFileVersionsRequestData = {
+    method: 'PROPFIND',
+    path: Pact.Matchers.regex({
+      matcher: `.*\\/remote\\.php\\/dav\\/meta\\/${fileInfo.id}\\/v$`,
+      generate: `/remote.php/dav/meta/${fileInfo.id}/v`
+    }),
+    headers: validAuthHeaders,
+    body: '<?xml version="1.0"?>\n' +
+      '<d:propfind  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">\n' +
+      '  <d:prop>\n' +
+      '  </d:prop>\n' +
+      '</d:propfind>'
+  }
 
-  function sleep (ms) {
-    return new Promise(resolve => setTimeout(resolve, ms))
+  const propfindFileVersionsResponse = (version, contentLength) => {
+    return `'    <d:response>\\n' +
+            \`        <d:href>/remote.php/dav/meta/${fileInfo.id}/v/${fileInfo.versions[version].versionId}</d:href>\\n\` +
+            '        <d:propstat>\\n' +
+            '            <d:prop>\\n' +
+            '                <d:getlastmodified>Thu, 08 Oct 2020 02:28:50 GMT</d:getlastmodified>\\n' +
+            \`                <d:getcontentlength>${contentLength}</d:getcontentlength>\\n\` +
+            '                <d:resourcetype/>\\n' +
+            '                <d:getetag>&quot;bc7012325dcc9899be7da7cabfdddb00&quot;</d:getetag>\\n' +
+            '                <d:getcontenttype>text/plain</d:getcontenttype>\\n' +
+            '            </d:prop>\\n' +
+            '            <d:status>HTTP/1.1 200 OK</d:status>\\n' +
+            '        </d:propstat>\\n' +
+            '        <d:propstat>\\n' +
+            '            <d:prop>\\n' +
+            '                <d:quota-used-bytes/>\\n' +
+            '                <d:quota-available-bytes/>\\n' +
+            '            </d:prop>\\n' +
+            '            <d:status>HTTP/1.1 404 Not Found</d:status>\\n' +
+            '        </d:propstat>\\n' +
+            '    </d:response>\\n' +`
```

Comment:
those ticks and quotes look strange
1. the whole return is in enclosed in ticks \` but also in quotes `'`
2. why do we need to have double escaped newlines `\\n` in other places we use only `\n `
3. why do we need the last `+` in line 70?